File: C_Programming/Practise_Programs/PP_174.c
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_174
//
//  Description       : This program turns OFF the 4th bit of a given
//                      unsigned integer using the bitwise AND
//                      operator and an appropriate bit mask.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 21/08/2026
//
//  Time Complexity   : O(1)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

#include<stdio.h>

typedef unsigned int UINT;

int main()
{
    UINT iNo = 0;
    UINT iMask = 0xFFFFFFF7;

    printf("Enter Number : \n");
    scanf("%u", &iNo);

    iNo = iNo & iMask;
    
    printf("Updated number : %u\n", iNo);

    return 0;
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      15
//
//  Sample Output :
//
//      Enter Number :
//      Updated number : 7
//
//////////////////////////////////////////////////////////////////

File: Java_Programming/Practise_Programs/PP_174.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_174
//
//  Description       : This program accepts a number from the user
//                      and displays an inverted right-aligned
//                      triangular pattern using star (*) symbols.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 05/09/2026
//
//  Time Complexity   : O(n^2)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

import java.util.Scanner;

class Pattern
{
    public static void Display(int iNo)
    {
        int i, j, k = 0;

        for(i = iNo; i >= 1; i--)
        {
            for(j = iNo - i; j >= 1; j--)
            {
                System.out.print(" \t");
            }            
            for(k = 1; k <= i; k++)
            {
                System.out.print("*\t");
            }
            System.out.println();
        }
    }
}

class PP_174
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int iValue = 0;

        System.out.println("Enter the number : ");
        iValue = sobj.nextInt();

        Pattern pobj = new Pattern();

        pobj.Display(iValue);
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      Enter the number :
//      5
//
//  Sample Output :
//      *   *   *   *   *
//          *   *   *   *
//              *   *   *
//                  *   *
//                      *
//
//////////////////////////////////////////////////////////////////